js/cards.js

```diff
@@ -114,6 +152,56 @@ export async function defineCardComponent() {
   customElements.define("tcg-card", Card);
 }
 
+function addAnimation(e){
+  // apply the animated class to start the CSS animation
+  e.target.classList.add("animated");
+}
+function updateZIndex(e){
+    e.target.style.zIndex = card_z_index;
+    // we don't want this to update again when reanimating later
+    e.target.removeEventListener("animationstart", updateZIndex);
+}
+
+function setupHover(){
+  // setup the initial event listeners for all cards
+  let cardList = document.getElementsByClassName("card-image");
+  for (let i = 0; i < cardList.length; i++) {
+    cardList[i].addEventListener("animationend", function(){
+      cardList[i].classList.remove("animated");
+    })
+    cardList[i].addEventListener("mouseover", addAnimation);
+    cardList[i].addEventListener("click", addAnimation);
+  }
+}
+export function addHover(){
+  // remove any unclickable classes from all the opened cards, so all can be interacted with
+  let clickedCard = document.getElementsByClassName("opened");
+  for (let i = 0; i < clickedCard.length; i++) {
+    clickedCard[i].classList.remove("unclickable");
+  }
+}
+export function removeHover(){
+  // check the current format is not grid
+  if(gacha_display_selection !== "gacha-grid"){
+
+    // this is the parent card container
+    let tcgCard = document.getElementsByClassName("clicked");
```

Comment:
rename to `allFlippedCards` - `tcgCard` suggests it's only one card and it's unclear what state it's in